python-connectors/googlesheets-sheet/connector.json

```diff
@@ -9,12 +9,34 @@
     "canCountRecords" : true,
 
     "params": [
+        {
+            "name": "auth_type",
+            "label": "Type of authentication",
+            "type": "SELECT",
+            "selectChoices": [
+                {
+                    "value": "service-account",
+                    "label": "Service account preset"
+                },
+                {
+                    "value": "credentials",
+                    "label": "Service account credentials (legacy)"
+                }
+            ]
+        },
         {
             "name": "credentials",
             "label": "Service Account credentials",
             "description": "Copy/paste the JSON private key provided by Google, or a path to a file containing the key. Click on the link above above to read how to get it.",
             "type": "TEXTAREA",
-            "mandatory" : true
+            "visibilityCondition": "model.auth_type == 'credentials' || model.auth_type.mode == 'NONE'"
```

Comment:
I *think* the second condition should mean I see this inout field on a legacy connection even when I haven't changed the auth type selection - but I wasn't seeing that. Maybe it needs a tweak to get it working?

Reply:
Found a way ! : [c74a3ff](https://github.com/dataiku/dss-plugin-googlesheets/pull/2/commits/c74a3ff1167f358ee1faa47f1c78a37b9512b0ee)

Reply:
Nice :thumbsup: